lua/entities/gmod_wire_expression2/core/string.lua

```diff
@@ -36,6 +36,71 @@ end)
 
 /******************************************************************************/
 
+local string_sub = string.sub
+registerOperator("fea", "nss", "", function(self, args)
+	local keyname, valname = args[2], args[3]
+	local str = args[4]
+	str = str[1](self, str)
+
+	local statement = args[5]
+
+	for key=1, #str do
+		local value = string_sub(str, key, key)
+		self:PushScope()
+
+		self.prf = self.prf + 1
```

Comment:
Nitpick but you could have this before the for loop with self.prf = self.prf + #str so it errors before doing any computation (or use #str saved to a var)

Reply:
Nice idea, but what about `break`?
```golo
foreach(N,Char:number = "counter-nitpick":repeat(999)){
    break
}
```
Your suggestion will incorrectly quota the E2 in cases like this.